chia/rpc/wallet_rpc_api.py

```diff
@@ -1289,91 +1306,56 @@ async def did_transfer_did(self, request):
     # NFT Wallet
     ##########################################################################################
 
-    async def nft_mint_nft(self, request):
-        wallet_id = int(request["wallet_id"])
+    async def nft_mint_nft(self, request) -> Dict:
+        log.debug("Got minting RPC request: %s", request)
+        wallet_id = uint32(request["wallet_id"])
+        assert self.service.wallet_state_manager
         nft_wallet: NFTWallet = self.service.wallet_state_manager.wallets[wallet_id]
-        address = request["artist_address"]
+        address = request.get("artist_address")
         if isinstance(address, str):
-            address = decode_puzzle_hash(address)
+            puzzle_hash = decode_puzzle_hash(address)
+        elif address is None:
+            puzzle_hash = await nft_wallet.standard_wallet.get_new_puzzlehash()
+        else:
+            puzzle_hash = address
```

Comment:
Is it possible to hit this condition? I would think that the address could only be either a string or None to successfully be marshalled through the RPC interface. I would raise an exception in this case.

Reply:
previous RPC was sending over bytes to pass puzzle hash and string to pass address, I added the None condition but it doesn't _need_ to be there, it's just the way I was writing unit tests for RPC

Reply:
I see. So it sounds like there are some RPC callers (maybe only in the tests) that don't go through JSON.